### DISCUSSION TITLE
Is there any way to completely hide all messages returned by the fit method for forecast task?

### DISCUSSION THREAD

**Discussion by good-morning-jeremy** [microsoft/FLAML#262] [Q&A]
Even though setting the `verbose = 0` in the `fit` method will minimize the output messages but there is still a wall of text that I find inconvenient because sometimes I don't really care about the searching process. Does anyone know how to hide it completely just like the `verbose = False` in Tensorflow?
These unavoidable messages are super annoying in my case when I have to combine the results of multiple different auto models in a single main function that includes many other steps :(

**Comment by sonichi** [comment]
Is the message like
>***/python3.7/site-packages/sklearn/linear_model/_sag.py:354: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge

That's the only message I get when using v0.6.9 and verbose=0. It's not from flaml but from sklearn. You can filter them by
```
import warnings
warnings.filterwarnings("ignore", module="sklearn[.*]")
```

**Reply by good-morning-jeremy** [reply]
Unfortunately, the message I got is way longer than that and it doesn't look like the warning from sklearn. I am doing the task `forecast` with these configurations:
```
SETTINGS = {
    'time_budget': 180,
    'task': 'forecast',
    'metrics': 'mape',
    'period': TEST_SIZE,
    'eval_method': 'holdout',
    'early_stop': True,
    'seed': 4,
    'n_jobs': -1,
    'verbose': 0,
}
```

This is how the message looks like and its actual length is 30 ~ 40 times of this: 
```
INFO:prophet:Disabling weekly seasonality. Run prophet with weekly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
Initial log joint probability = -2.31746
    Iter      log prob        ||dx||      ||grad||       alpha      alpha0  # evals  Notes 
      99         258.4   1.19501e-07        492.94      0.7939      0.7939      124   
    Iter      log prob        ||dx||      ||grad||       alpha      alpha0  # evals  Notes 
     147       258.401   8.20591e-08       405.836   1.727e-10       0.001      224  LS failed, Hessian reset 
     157       258.401   5.75377e-09       486.971      0.4539      0.4539      236   
Optimization terminated normally: 
  Convergence detected: absolute parameter change was below tolerance
INFO:prophet:Disabling weekly seasonality. Run prophet with weekly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
Initial log joint probability = -2.31746
    Iter      log prob        ||dx||      ||grad||       alpha      alpha0  # evals  Notes 
      99       259.758   2.60814e-05       590.255           1           1      123   
    Iter      log prob        ||dx||      ||grad||       alpha      alpha0  # evals  Notes 
     123       259.762   8.73681e-09       604.386           1           1      155   
Optimization terminated normally: 
  Convergence detected: absolute parameter change was below tolerance
INFO:prophet:Disabling weekly seasonality. Run prophet with weekly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
Initial log joint probability = -2.31746
    Iter      log prob        ||dx||      ||grad||       alpha      alpha0  # evals  Notes 
      99       260.188   1.94398e-08       463.025      0.1679      0.1679      124   
    Iter      log prob        ||dx||      ||grad||       alpha      alpha0  # evals  Notes 
     106       260.188   7.73027e-09       487.398      0.3107      0.3107      133   
Optimization terminated normally: 
  Convergence detected: absolute parameter change was below tolerance
INFO:prophet:Disabling weekly seasonality. Run prophet with weekly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
```

**Reply by sonichi** [reply]
That's from prophet. You can follow https://stackoverflow.com/questions/45551000/how-to-control-output-from-fbprophet to suppress the output.
@int-chaos for awareness.

**Reply by good-morning-jeremy** [reply]
It works like a charm, thank you Sonichi!

Just a little further step to completely ignore all messages from Prophet. 
If you want to suppress this message: 
`INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.`
Which can add up a lot if your search is long then you can do it by setting the logging level higher than INFO for `prophet`:
```
import logging
logging.getLogger('prophet').setLevel(logging.WARNING) 
```

By combining this and your above solution, it is truly `verbose = 0`.

**Reply by sonichi** [reply]
@good-morning-jeremy Thanks. What application do you use flaml for? Feel free to chat on [gitter](https://gitter.im/FLAMLer/community).
